### PR TITLE
Update portage

### PR DESCRIPTION
**Pull request by dm0-**
Depends on coreos/portage-stable#679

### CODE REVIEW

File: sys-apps/portage/files/portage-2.3.8-0002-backport-decompression-fixes.patch [hunk 1]
```diff
@@ -1,42 +0,0 @@
...
--			"echo tar failed with status ${p[1]} ; exit ${p[1]} ; fi ; " + \
-+			("cmd0=(head -c-%d -- %s) cmd1=(%s) cmd2=(tar -xp %s -C %s -f -); " + \
-+			'"${cmd0[@]}" | "${cmd1[@]}" | "${cmd2[@]}"; ' + \
-+			"p=(${PIPESTATUS[@]}) ; for i in {0..2}; do " + \
-+			"if [[ ${p[$i]} != 0 && ${p[$i]} != %d ]] ; then " + \
-+			"echo command $(eval \"echo \\\"'\\${cmd$i[*]}'\\\"\") " + \
-+			"failed with status ${p[$i]} ; exit ${p[$i]} ; fi ; done; " + \
-+			"if [ ${p[$i]} != 0 ] ; then " + \
-+			"echo command $(eval \"echo \\\"'\\${cmd$i[*]}'\\\"\") " + \
-+			"failed with status ${p[$i]} ; exit ${p[$i]} ; fi ; " + \
```
Comment (cgwalters) [inline sys-apps/portage/files/portage-2.3.8-0002-backport-decompression-fixes.patch]: Wow.

File: sys-apps/portage/files/portage-2.2.18-0001-portage-repository-config.py-add-disabled-attribute-.patch [hunk 2]
```diff
@@ -1,51 +0,0 @@
-From 1b8ccc0e487a523aa6d974b3ebe26548ad6502b6 Mon Sep 17 00:00:00 2001
```
Comment (ajeddeloh) [inline sys-apps/portage/files/portage-2.2.18-0001-portage-repository-config.py-add-disabled-attribute-.patch]: this is safe to drop with the new portage?

Comment (dm0-) [reply sys-apps/portage/files/portage-2.2.18-0001-portage-repository-config.py-add-disabled-attribute-.patch]: It's dropping the `2.2.18` patch; the new version is using the `2.3.8` patch.

**Comment by dm0-** [comment]
This portage update causes the toolchains bootstrapping script to fail, so #3313 was used as a workaround until the script can be fixed.

**Pull request merged**